webapp/channels/src/actions/views/create_comment.tsx

```diff
@@ -103,6 +105,25 @@ export function submitPost(channelId: string, rootId: string, draft: PostDraft,
 
         post = hookResult.data;
 
+        if (schedulingInfo) {
```

Comment:
Perhaps instead of having to pass `schedulingInfo` all the way through `onSubmit` and `submitPost`, we could instead reorganize this so that all of the checking for channel mentions and running hooks are done by one or more helper functions that we can call from both `submitPost` and some new `schedulePost`? Moving those into helper functions that can be called by two different actions is some extra work, but it'll help make it easier to understand what all happens when you click the submit buttons.

I want to avoid the type of issue we run into with the `Post` component where it's full of complex if statements because we went in the direction of a single `Post` component that can behave a variety of different ways due to various combinations of parameters instead of making multiple components like `RhsPost`, `CenterChannelPost`, and `SearchPost` that share children. In both that case and this one, I think it's better to have two single purpose pieces of code that use the same helpers in the same order rather than to add conditionals like that since you get code with fewer pieces of branching logic that need to be understood.

Thoughts? I'm open to discussing this as well since I'm aware that I'm also asking you to help clean up a bunch of existing code

Reply:
I've gone through this discussion with Daniel and what we have now is the cleanest version possible. I initially did start by implementing a different hook for processing scheduled post, but if as you can see, all we need to do differently is to skip just 3-4 steps in the whole chain of function calls. Splitting this into helper functions and having two separate hooks uses those helpers sounds good, but I am sure it will cause the issue of someone making a change in send post logic without testing and making the same change in schedule post logic. Current code avoid that issue.

If you go through the whole function chain, you'll realise its not a lot of if checks and there is no mixing of logic.

Reply:
That makes sense to me. I do think reducing the complexity of that flow might be worth the tradeoff of making it so that people need to add things in two places, but since most of that would require reorganizing a lot of existing code, that's definitely outside the scope of what we want to do here